subprojects/cfa/cfa/src/main/java/hu/bme/mit/theta/cfa/dsl/CfaExpression.java

```diff
@@ -705,17 +757,30 @@ private BvConcatExpr createBvConcatExpr(final Expr<BvType> leftOp, final Expr<Bv
 
 		////
 
+		@SuppressWarnings("unchecked")
 		@Override
 		public Expr<?> visitUnaryExpr(final UnaryExprContext ctx) {
 			if (ctx.op != null) {
 				final Expr<?> op = ctx.op.accept(this);
-				switch(ctx.oper.getType()) {
+				switch (ctx.oper.getType()) {
 					case PLUS:
 						return Pos(op);
 
 					case MINUS:
 						return Neg(op);
 
+					case FP_ABS:
+						return Abs((Expr<FpType>) op);
+
+					case FP_IS_NAN:
+						return IsNan((Expr<FpType>) op);
+
+					case FPROUNDTOINT:
+						return RoundToIntegral(FpRoundingMode.RNE, (Expr<FpType>) op);
+
+					case FPSQRT:
+						return Sqrt(FpRoundingMode.RNE, (Expr<FpType>) op);
```

Comment:
```suggestion
						return Sqrt(FpRoundingMode.getDefaultRoundingMode(), (Expr<FpType>) op);
```

Reply:
Fixed in [19a3e6b](https://github.com/ftsrg/theta/pull/123/commits/19a3e6b5a7cf198aa7c57c6ee2fd3aec2a75a257)